drivers/display/display_st7789v.c

```diff
@@ -288,54 +303,58 @@ static void st7789v_lcd_init(const struct device *dev)
 	tmp = 0x0f;
 	st7789v_transmit(dev, ST7789V_CMD_FRCTRL2, &tmp, 1);
 
-	tmp = DT_INST_PROP(0, gctrl);
+	tmp = config->gctrl;
 	st7789v_transmit(dev, ST7789V_CMD_GCTRL, &tmp, 1);
 
-	tmp = DT_INST_PROP(0, vcom);
+	tmp = config->vcom;
 	st7789v_transmit(dev, ST7789V_CMD_VCOMS, &tmp, 1);
 
-#if (DT_INST_NODE_HAS_PROP(0, vrhs) && \
-	DT_INST_NODE_HAS_PROP(0, vdvs))
-	tmp = 0x01;
-	st7789v_transmit(dev, ST7789V_CMD_VDVVRHEN, &tmp, 1);
+	if (config->vdv_vrh_enable) {
+		tmp = 0x01;
+		st7789v_transmit(dev, ST7789V_CMD_VDVVRHEN, &tmp, 1);
 
-	tmp = DT_INST_PROP(0, vrhs);
-	st7789v_transmit(dev, ST7789V_CMD_VRH, &tmp, 1);
+		tmp = config->vrh_value;
+		st7789v_transmit(dev, ST7789V_CMD_VRH, &tmp, 1);
 
-	tmp = DT_INST_PROP(0, vdvs);
-	st7789v_transmit(dev, ST7789V_CMD_VDS, &tmp, 1);
-#endif
+		tmp = config->vdv_value;
+		st7789v_transmit(dev, ST7789V_CMD_VDS, &tmp, 1);
+	}
 
-	st7789v_transmit(dev, ST7789V_CMD_PWCTRL1, st7789v_pwctrl1_param,
-			 sizeof(st7789v_pwctrl1_param));
+	memcpy(tmp_buf, config->pwctrl1_param, sizeof(config->pwctrl1_param));
+	st7789v_transmit(dev, ST7789V_CMD_PWCTRL1, tmp_buf,
+			 sizeof(config->pwctrl1_param));
 
 	/* Memory Data Access Control */
-	tmp = DT_INST_PROP(0, mdac);
+	tmp = config->mdac;
 	st7789v_transmit(dev, ST7789V_CMD_MADCTL, &tmp, 1);
 
 	/* Interface Pixel Format */
-	tmp = DT_INST_PROP(0, colmod);
+	tmp = config->colmod;
 	st7789v_transmit(dev, ST7789V_CMD_COLMOD, &tmp, 1);
 
-	tmp = DT_INST_PROP(0, lcm);
+	tmp = config->lcm;
 	st7789v_transmit(dev, ST7789V_CMD_LCMCTRL, &tmp, 1);
 
-	tmp = DT_INST_PROP(0, gamma);
+	tmp = config->gamma;
 	st7789v_transmit(dev, ST7789V_CMD_GAMSET, &tmp, 1);
 
 	st7789v_transmit(dev, ST7789V_CMD_INV_ON, NULL, 0);
 
-	st7789v_transmit(dev, ST7789V_CMD_PVGAMCTRL, st7789v_pvgam_param,
-			 sizeof(st7789v_pvgam_param));
+	memcpy(tmp_buf, config->pvgam_param, sizeof(config->pvgam_param));
```

Comment:
Do not bother with "warning: passing argument 3 of 'blabla' discards 'const'", just:
`st7789v_transmit(dev, ST7789V_CMD_PVGAMCTRL, (void *)config->pvgam_param, sizeof(config->pvgam_param));`

Reply:
I added the const conversion as requested.